5watchdog

```diff
@@ -158,8 +158,28 @@ fi
 ## Check internet before starting main loop
 while ! nc -vzw1 google.com 443;
 do
-  echo "$(date) - [WARNING] - Internet is down, checking again in 10 seconds..."
+  DNS=0
+  echo "$(date) - [WARNING] - Internet is down, checking ..."
   sleep 10
+  if ping -c1 8.8.8.8 &> /dev/null
+  then
+    echo "$(date) - [WARNING] - Ping ok, but problem in name resolution: checking every 30 sec" | tee -a ${LOG_FILE}
+    while  ! nc -vzw1 google.com 443  && (( DNS < 20 ))
+    do
+      echo "Name resolution not succeful yet"
+      sleep 30
+      DNS=$((DNS + 1))
+    done
+    if ping -c1 8.8.8.8 &> /dev/null && ! nc -vzw1 google.com 443
+    then
+      echo " Name resolution problem while ping is ok, waited 10 minutes, restarting system in 10 sec"
+      sleep 10
+      sudo reboot
+    elif ping -c1 8.8.8.8 &> /dev/null && nc -vzw1 google.com 443
+    then
+      echo "Ping and DNS are OK"
+    fi
+  fi
```

Comment:
uhm… it is far too complex, simplify this by doing:
```
DNS=0
while ! nc …
do
  if (( DNS >= 20 ))
  then
    echo … waited 10 minutes ...
    reboot
    exit
  fi
  echo … failed test connection ...
  if ping …
  then
    echo … but ping is ok ...
    (( DNS++ ))
  fi
  sleep ...
done
```

Reply:
👍